src/core/dom/layout/viewport-observer.js

```diff
@@ -81,7 +77,12 @@ export function unobserveWithSharedInOb(element) {
  */
 function ioCallback(entries) {
   for (let i = 0; i < entries.length; i++) {
-    const {isIntersecting, target} = entries[i];
-    viewportCallbacks.get(target)?.(isIntersecting);
+    const entry = entries[i];
+    const {target} = entry;
+    const callbacks = viewportCallbacks.get(target);
+    for (let j = 0; j < callbacks.length; j++) {
+      const callback = callbacks[j];
+      callback(entry);
+    }
```

Comment:
FMI: Why doesn't this logic only call the callbacks for the last entry corresponding a given target? It looks like the callbacks are being called for each entry.

Reply:
Good point. That's what we had done before using this viewport observer in the custom hook. @samouri was this something you considered when implementing this? Any issues if we change this logic to only call the last entry?

Reply:
yep! sounds more correct go to in reverse order and skip dupes.
rn i'm more concerned about changing the ioCallback signature (current usages of observeWithSharedInOb will break)

Reply:
Done.